src/index/updater.rs

```diff
@@ -630,6 +679,60 @@ impl<'index> Updater<'index> {
     Ok(())
   }
 
+  fn index_transaction_addresses(
+    &mut self,
+    tx: &Transaction,
+    txid: &Txid,
+    txout_receiver: &mut broadcast::Receiver<TxOut>,
+    utxo_cache: &mut HashMap<OutPoint, TxOut>,
+    script_pubkey_to_outpoint: &mut MultimapTable<&[u8], OutPointValue>,
+    outpoint_to_txout: &mut Table<&OutPointValue, TxOutValue>,
+    index_inscriptions: bool,
+  ) -> Result {
+    for txin in &tx.input {
+      let output = txin.previous_output;
+      if output.is_null() {
+        continue;
+      }
+
+      // multi-level cache for UTXO set to get to the script pubkey
+      let txout = if let Some(txout) = utxo_cache.get(&txin.previous_output) {
+        txout.clone()
+      } else if let Some(value) = outpoint_to_txout.get(&txin.previous_output.store())? {
+        TxOut::load(value.value())
+      } else {
+        txout_receiver.blocking_recv().map_err(|_| {
```

Comment:
maybe add error message from channel in here
```suggestion
        txout_receiver.blocking_recv().map_err(|err| {
```